source/integrate-with-integration-environment/authenticate-your-user.html.md.erb

```diff
@@ -369,6 +369,92 @@ You can use the following table to understand the ID token’s claims.
+
+1. Download the JWKS document from the JWKS endpoint. This endpoint can be found as the `jwks_uri` value on the discovery endpoint: 
+  * Integration environment: [https://oidc.integration.account.gov.uk/.well-known/openid-configuration](https://oidc.integration.account.gov.uk/.well-known/openid-configuration) 
+  * Production environment: [https://oidc.account.gov.uk/.well-known/openid-configuration](https://oidc.account.gov.uk/.well-known/openid-configuration).
+1. Find the object in `keys` which has a `kid` field matching the `kid` from the JWT header in the ID token. 
+1. You must [validate that the JWT `alg` header matches](https://datatracker.ietf.org/doc/html/rfc8725#section-3.1) what was returned from the `jwks_uri`.
+1. Use the key to verify the signature of the ID token. 
+
+
+<%= warning_text('If the `kid` in the JWT header does not match a `kid` in the JWKS document, you must not trust that ID token. You should contact GOV.UK One Login through the <a href="https://www.sign-in.service.gov.uk/contact-us">contact us form</a>, including the name of your service and client ID.') %>
+
+#### Caching the JWKS document 
+
+You should cache the returned JWKS document and reuse it instead of downloading the JWKS document for every signature you need to verify. The JWKS document will not regularly change and caching it reduces latency for your service.
+
+The `Cache-Control` HTTP header field in the JWKS endpoint contains a suggested caching period. This caching period is the time period that GOV.UK One Login expects the JWKS document to remain valid. 
+
+For example, a header with the value `Cache-Control: max-age=86400` would mean you cache the JWKS document for the `max-age` of 1 day (86400 seconds).
+
+Occasionally, if there’s a temporary outage, you may not be able to refresh the cache from GOV.UK One Login’s JWKS endpoint. If this happens, you should continue to trust the cached version until you’re able to refresh the cache.
+
+For more details on the `Cache-Control` header, see [RFC 9111: HTTP Caching](https://www.rfc-editor.org/rfc/rfc9111#field.cache-control). 
+
+#### Understanding the ID token signing key rotations
+
+GOV.UK One Login will rotate the keys as follows:
+
+- In the integration environment -  in Jan 2026 and Feb 2026 to enable you to test your integration.
+- In the production environment - every 6 months starting from Mar 2026. 
+- In some scenarios we may rotate keys at short notice.
+
+New public keys will appear in the `keys` array of the JWKS document before any rotation.
+
+Use the `Cache-Control` headers and [guidance on caching the JWKS document](/integrate-with-integration-environment/authenticate-your-user/#caching-the-jwks-document) to regularly poll the JWKS endpoint to detect new versions and make sure you’re using the latest key available.
+
+Once GOV.UK One Login has removed the old public key from the JWKS document, it will no longer be valid. You should no longer trust ID tokens signed with that key.
```

Comment:
```suggestion
Once GOV.UK One Login has removed the old public key from the JWKS document, it will no longer be valid. Do not trust ID tokens signed with that key.
```